newdle/api.py

```diff
@@ -181,14 +190,20 @@ def get_my_newdles():
 @api.route('/newdle/', methods=('POST',))
 @use_kwargs(NewNewdleSchema(), locations=('json',))
 def create_newdle(title, duration, timezone, timeslots, participants):
+    participants = [(p.pop('signature'), Participant(**p)) for p in participants]
+
+    for signature, p in participants:
+        if not check_user_signature(p.auth_uid, p.email, signature):
+            raise BadRequest
+
     newdle = Newdle(
         title=title,
         creator_uid=g.user['uid'],
         creator_name=f'{g.user["first_name"]} {g.user["last_name"]}',
         duration=duration,
         timezone=timezone,
         timeslots=timeslots,
-        participants={Participant(**p) for p in participants},
+        participants=set(p[1] for p in participants),
```

Comment:
`{p[1] for p in participants}` should work as well

Reply:
Indeed.